src/fuse_reduce.cpp

```diff
@@ -389,8 +411,8 @@ void fuse_reduce::apply(module_pass_manager& mpm) const
     {
         if(enable_rewrite_reshapes)
             mpm.run_pass(rewrite_reshapes<reduce_reshape>{});
-        match::find_matches(
-            mpm, find_reduce_pointwise{}, find_pointwise_reduce{}, find_reduce_reduce{});
+        match::find_matches(mpm, find_reduce_pointwise{}, find_pointwise_reduce{});
+        match::find_matches(mpm, find_reduce_reduce{});
```

Comment:
Why is this moved out?

Reply:
because when it hits the broadcast condition, its not fusing the other fused_reduce in the input

Reply:
We might need to add this to the matcher then.